runtime/src/main/java/org/corfudb/runtime/view/stream/IStreamView.java

```diff
@@ -227,15 +230,15 @@ default ILogData next() {
      */
     long getTotalUpdates();
 
-    /**
-     * Get the compaction mark of this stream.
-     * @return compaction mark.
-     */
     long getCompactionMark();
 
-    /**
-     * Set the compaction mark of this stream.
-     * @param compactionMark updated compaction mark address.  
-     */
-    void setCompactionMark(long compactionMark);
+    default ILogData streamReadWithCheck(Supplier<ILogData> dataFetcher, long snapshot) {
```

Comment:
This is not a good name can you change it?

Reply:
This function is replaced by another function called compactionMarkCheck.